File: 01/solution_B.py
# depths = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263]  # example

with open("depths.txt") as depths_file:
    depths = [*map(int, depths_file.readlines())]

increases = 0

previous_window_sum = sum(depths[0:3])
for i in range(len(depths) - 2):
    window_sum = sum(depths[i:i + 3])
    if window_sum > previous_window_sum:
        increases += 1
    previous_window_sum = window_sum

print(increases)

File: 01/solution_B.cpp
#include <cstdio>
#include <stdexcept>

#include "../common/readlines.hpp"


void print_usage(char name[]) {
    printf("USAGE: %s INPUT.TXT\n", name);
}


int main(int argc, char* argv[]) {
    if (argc != 2) {
        print_usage(argv[0]);
        return 0;
    }
    int depths[2000];
    int depths_count = readlines_int(argv[1], depths);
    int increases = 0;  // <- solution
    int last_window_sum = depths[0] + depths[1] + depths[2];
    for (int i = 1; i < depths_count - 2; i++) {
        int window_sum = last_window_sum - depths[i - 1] + depths[i + 2];
        if (window_sum > last_window_sum) {
            increases++;
        }
        last_window_sum = window_sum;
    }
    printf("Depth window increases %d times\n", increases);
    return 0;
}
